doc/packet_trimming/packet-trimming-design.md

```diff
@@ -0,0 +1,783 @@
+
+counterpoll port interval 1000
+counterpoll queue interval 1000
+counterpoll acl interval 1000
+```
+
+#### 2.6.2.2 Show command group
+
+**The following command shows switch trimming global configuration:**
+```bash
+root@sonic:/home/admin# show switch-trimming global
++---------------------------------------+---------+
+| Configuration                         | Value   |
++=======================================+=========+
+| Packet trimming size                  | 128     |
++---------------------------------------+---------+
+| Packet trimming DSCP value            | 48      |
++---------------------------------------+---------+
+| Packet trimming queue index           | 6       |
++---------------------------------------+---------+
+
+root@sonic:/home/admin# show switch-trimming global --json
+{
+    "size": "128",
+    "dscp_value": "48",
+    "queue_index": "6"
+}
+```
+
+**The following command shows switch trimming buffer configuration:**
+```bash
+root@sonic:/home/admin# show buffer configuration
+Lossless traffic pattern:
+------------------  -
+default_dynamic_th  0
+------------------  -
+
+Pool: egress_lossy_pool
+----  -------
+mode  dynamic
+type  egress
+----  -------
+
+Profile: q_lossy_trim_profile
+---------------------  -----------------
+dynamic_th             3
+packet_discard_action  trim
+pool                   egress_lossy_pool
+size                   0
+---------------------  -----------------
+```
+
+**The following command shows switch trimming ACL policy configuration:**
+```bash
+root@sonic:/home/admin# show acl table
+Name        Type         Binding    Description      Stage    Status
+----------  -----------  ---------  ---------------  -------  --------
+TRIM_TABLE  TRIMMING_L3  Ethernet0  Packet trimming  ingress  Active
+
+root@sonic:/home/admin# show acl rule
+Table       Rule         Priority  Action        Match               Status
+----------  ---------  ----------  ------------  ------------------  --------
+TRIM_TABLE  TRIM_RULE         999  DISABLE_TRIM  SRC_IP: 1.1.1.1/32  Active
+```
+
+**The following command shows switch trimming counter configuration:**
+```bash
+root@sonic:/home/admin# counterpoll show
+Type          Interval (in ms)    Status
+------------  ------------------  --------
+PORT_STAT     1000                enable
+QUEUE_STAT    1000                enable
+ACL           1000                enable
+```
+
+**The following command shows switch trimming statistics:**
+```bash
+root@r-r640-03-135:/home/admin# portstat -i Ethernet0
+    IFACE    STATE    RX_OK    RX_BPS    RX_UTIL    RX_ERR    RX_DRP    RX_OVR    TX_OK    TX_BPS    TX_UTIL    TX_ERR    TX_DRP    TX_OVR    TRIM
+---------  -------  -------  --------  ---------  --------  --------  --------  -------  --------  ---------  --------  --------  --------  ------
+Ethernet0        U      100  0.00 B/s      0.00%         0         0         0        0  0.00 B/s      0.00%         0       100         0     100
```

Comment:
Showing TRIM stats on platforms where it is NOT supported is unnecessarily bloating the output of a very frequently used CLI.

Maybe display it only when "show interfaces counters detailed Ethernet0" and/or the below "portstat --trim" option.

Example: https://github.com/sonic-net/SONiC/blob/master/doc/qos/ECN_and_WRED_statistics_HLD.md#show-interface-counters-cli-output-on-a-wred-drop-statistics-supported-platform

Reply:
@rck-innovium done